### PR TITLE
feat: Added Cache API

### PR DESCRIPTION
**Pull request by JigarJoshi**
Initial import of Tigris Cache API.

### CODE REVIEW

File: server/v1/cache.proto [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+      delete : "/v1/projects/{project}/caches/{name}/delete"
+      body : "*"
+    };
+    option(openapi.v3.operation) = {
+      summary: "Creates the cache"
+      tags: "Cache"
+    };
+  }
+
+  rpc Put(PutRequest) returns (PutResponse) {
```
Comment (himank) [inline server/v1/cache.proto:132]: Better to use Cache terminology here, use `Set`

File: server/v1/cache.proto [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+}
+
+message DelResponse {
+  // A detailed response message.
+  string message = 1;
+  // An enum with value set as "deleted" or "not_found"
+  string status = 2;
+}
+
+message KeysRequest {
```
Comment (himank) [inline server/v1/cache.proto:97]: We can have `pattern` in the options.

File: server/v1/cache.proto (hunk 1 above)
Comment (JigarJoshi) [reply server/v1/cache.proto]: updated

File: server/v1/cache.proto [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+}
+
+message DeleteCacheResponse {
+  // A detailed response message.
+  string message = 1;
+  // An enum with value set as "deleted"
+  string status = 2;
+}
+
+message SetRequest {
```
Comment (himank) [inline server/v1/cache.proto:45]: We can have similar options https://redis.io/commands/set/

File: server/v1/cache.proto (hunk 2 above)
Comment (JigarJoshi) [reply server/v1/cache.proto:97]: added

File: server/v1/cache.proto (hunk 3 above)
Comment (JigarJoshi) [reply server/v1/cache.proto:56]: Added some options. we can iterate on richer set of options.

**Review by himank — APPROVED** [review]

**Comment by ovaistariq** [comment]
Let's have the APIs aligned with what Redis has so that it is easier for users to adopt our cache.

**Pull request merged**